apis/nucleus/src/viz.js

```diff
@@ -149,6 +150,41 @@ export default function viz({ model, halo, initialError, onDestroy = async () =>
       const propertyTree = await conversionConvertTo({ halo, model, cellRef, newType });
       return propertyTree;
     },
+    /**
+     * Toggles the chart to a data view of the chart.
+     *
+     * The chart will be toggled to the type defined in the nebula context (dataViewType).
+     *
+     * The default dataViewType for nebula is sn-table. The specified chart type needs to be registered as well in to make it possible to render the data view.
+     *
+     * @experimental
+     * @since 4.7.0
+     * @param {boolean=} showDataView - If included, turns the toggle into a one way-only operation. If true it will only toggle to the view data table.
+     */
+    async toggleDataView(showDataView) {
+      let newModel;
+      if (!viewDataObjectId && showDataView !== false) {
+        const oldProperties = await model.getEffectiveProperties();
+        const propertyTree = await conversionConvertTo({
+          halo,
+          model,
+          cellRef,
+          newType: halo.config.context.dataViewType,
+          properties: oldProperties,
+          convertToTable: true,
+        });
+        const newProperties = { ...propertyTree.qProperty, totals: { show: false }, usePagination: true };
```

Comment:
feels questionable to add sn-table specific properties here.
(but I am unsure where to put it instead, maybe in a importViewDataProperties? in sn-table? or dataViewExtraProperties setting)

Reply:
That would probably help to keep this consistent between sense client and nebula and a good place to have it as source. @Caele do you agree on this?

Reply:
Probably makes more sense to have in sn-table. As currently nebula is dependent on some sn-table props that may or may not change in the future. So this logic in nebula may end up not being compatible with sn-table anymore because sn-table changed its properties.

Reply:
Ex: in version x.x.x of sn-table, we also want to use `someNewProp: "value"` for the view data case. If the user have a version where nebula does not have `someNewProp` set but does have sn-table x.x.x, viewData might not behave as expected

Reply:
So we could have a `viewDataMode` prop or somthing that would currently enable pagination and disable totals?

Reply:
No I think more like @T-Wizard suggested where we add a function to sn-table (e.g. importViewDataProperties) which provides these properties. Similar to the other export and import properties functionality

Reply:
I see, that would be nice

Reply:
yea, should avoid placing them here.
I could see having them just in importProperties, as defaults if there are no table props on the object already. To avoid adding anothing API specific to view data.